pennylane_lightning/core/src/simulators/lightning_gpu/measurements/MeasurementsGPU.hpp

```diff
@@ -0,0 +1,762 @@
+     */
+    template <class index_type>
+    auto expval(const index_type *csrOffsets_ptr,
+                const index_type csrOffsets_size, const index_type *columns_ptr,
+                const std::complex<PrecisionT> *values_ptr,
+                const index_type numNNZ) -> PrecisionT {
+        const std::size_t nIndexBits = this->_statevector.getNumQubits();
+        const std::size_t length = std::size_t{1} << nIndexBits;
+
+        auto device_id =
+            this->_statevector.getDataBuffer().getDevTag().getDeviceID();
+        auto stream_id =
+            this->_statevector.getDataBuffer().getDevTag().getStreamID();
+
+        std::unique_ptr<DataBuffer<CFP_t>> d_sv_prime =
+            std::make_unique<DataBuffer<CFP_t>>(length, device_id, stream_id,
+                                                true);
+
+        cuUtil::SparseMV_cuSparse<index_type, PrecisionT, CFP_t>(
+            csrOffsets_ptr, csrOffsets_size, columns_ptr, values_ptr, numNNZ,
+            this->_statevector.getData(), d_sv_prime->getData(), device_id,
+            stream_id, this->_statevector.getCusparseHandle());
+
+        auto expect =
+            innerProdC_CUDA(this->_statevector.getData(), d_sv_prime->getData(),
+                            this->_statevector.getLength(), device_id,
+                            stream_id, this->_statevector.getCublasCaller())
+                .x;
+
+        return expect;
+    }
+
+    /**
+     * @brief Expected value of an observable.
+     *
+     * @param operation String with the operator name.
+     * @param wires Wires where to apply the operator.
+     * @return Floating point expected value of the observable.
+     */
+    auto expval(const std::string &operation, const std::vector<size_t> &wires)
+        -> PrecisionT {
+        std::vector<PrecisionT> params = {0.0};
+        std::vector<ComplexT> gate_matrix = {};
+        return this->expval_(operation, wires, params, gate_matrix);
+    }
+
+    /**
+     * @brief Expected value for a list of observables.
+     *
+     * @tparam op_type Operation type.
+     * @param operations_list List of operations to measure.
+     * @param wires_list List of wires where to apply the operators.
+     * @return Floating point std::vector with expected values for the
+     * observables.
+     */
+    template <typename op_type>
+    auto expval(const std::vector<op_type> &operations_list,
+                const std::vector<std::vector<size_t>> &wires_list)
+        -> std::vector<PrecisionT> {
+        PL_ABORT_IF(
+            (operations_list.size() != wires_list.size()),
+            "The lengths of the list of operations and wires do not match.");
+        std::vector<PrecisionT> expected_value_list;
+
+        for (size_t index = 0; index < operations_list.size(); index++) {
+            expected_value_list.emplace_back(
+                expval(operations_list[index], wires_list[index]));
+        }
+
+        return expected_value_list;
```

Comment:
I think we want coverage here. Can we add a test that fixes the warning?

Reply:
hmmm, it's tested [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/5f20309098d83616f989c5581b359a6eccf27a74/pennylane_lightning/core/src/simulators/lightning_gpu/measurements/tests/Test_StateVectorCudaManaged_Measure.cpp#L222-L225). It's strange that codecov didn't capture it.